src/integrationTest/java/uk/gov/hmcts/reform/bulkscanprocessor/entity/EnvelopeRepositoryTest.java

```diff
@@ -89,6 +90,25 @@ public void findByZipFileName_should_find_envelops_in_db() {
         assertThat(resultForX).hasSize(0);
     }
 
+    @Test
+    public void should_get_1_incomplete_envelope_from_yesterday() {
+        // given
+        dbHas(
+            envelope("A.zip", "X", Status.PROCESSED),
+            envelope("B.zip", "Y", Status.COMPLETED),
+            envelope("C.zip", "Z", Status.UPLOAD_FAILURE),
+            envelope("D.zip", "Z", Status.COMPLETED)
+        );
+
+        // and update createAt to yesterday
+        entityManager.createNativeQuery(
```

Comment:
you could  just overload a method in envelope creator which accepts date field

Reply:
created at is final - not changeable